File: backend/src/main/java/com/tribly/dto/device/response/DeviceTokenResponse.java
package com.tribly.dto.device.response;

import com.tribly.dto.validation.ValidateSchema;
import lombok.Builder;
import org.eclipse.microprofile.openapi.annotations.media.Schema;
import org.jspecify.annotations.Nullable;

@Schema(description = "Device OAuth token response")
@Builder
@ValidateSchema
public record DeviceTokenResponse(
    @Schema(description = "Access token", required = true) String accessToken,
    @Schema(description = "Token type (always 'Bearer')", required = true) String tokenType,
    @Schema(description = "Token expiry in seconds", required = true) int expiresIn,
    @Schema(description = "Refresh token") @Nullable String refreshToken) {}

File: frontend/src/api/dto/deviceTokenResponse.ts
/**
 * Device OAuth token response
 */
export interface DeviceTokenResponse {
  /** Access token */
  accessToken: string
  /** Token type (always 'Bearer') */
  tokenType: string
  /** Token expiry in seconds */
  expiresIn: number
  /** Refresh token */
  refreshToken?: string
}
